crates/chia-datalayer/src/merkle.rs

```diff
@@ -398,8 +398,12 @@ fn get_free_indexes_and_keys_values_indexes(
 #[derive(Debug)]
 pub struct MerkleBlob {
     blob: Vec<u8>,
+    // TODO: would be nice for this to be deterministic ala a fifo set
     free_indexes: HashSet<TreeIndex>,
```

Comment:
can this be a `Vec<TreeIndex>`? I would expect all you do is push and pop indices, no?

Reply:
also membership checks.  i guess the _free_ indexes are likely often a small group but i was looking into a more complete solution.